packages/components/src/base-field/hook.js

```diff
@@ -0,0 +1,74 @@
+/**
+ * External dependencies
+ */
+import { cx } from 'emotion';
+
+/**
+ * WordPress dependencies
+ */
+import { useMemo } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { useContextSystem } from '../ui/context';
+import { useControlGroupContext } from '../ui/control-group';
+import { useFlex } from '../flex';
+import * as styles from './styles';
+
+/**
+ * @typedef OwnProps
+ * @property {boolean} [error=false] Renders an error.
+ * @property {boolean} [disabled] Whether the field is disabled.
+ * @property {boolean} [isClickable=false] Renders a `cursor: pointer` on hover.
+ * @property {boolean} [isFocused=false] Renders focus styles.
+ * @property {boolean} [isInline=false] Renders as an inline element (layout).
+ * @property {boolean} [isSubtle=false] Renders a subtle variant.
+ */
+
+/** @typedef {import('../Flex/useFlex').FlexProps & OwnProps} Props */
+
+/**
+ * @param {import('@wp-g2/create-styles').ViewOwnProps<Props, 'div'>} props
+ */
+export function useBaseField( props ) {
+	const {
+		className,
+		error = false,
+		isClickable = false,
+		isFocused = false,
```

Comment:
Is there any particular reason why we're using custom JS logic for focus states, instead of using the CSS `:focus` selector?

Reply:
@diegohaz do you have any ideas for if there are use cases for custom focus states? FWIW the styles cover both the manual state and the regular `:focus` state.

Reply:
Unless we want to mimic some custom focus behavior like `:focus-visible` (for browsers that still don't support it), which is not the case here, I'd say it's not necessary. If we want to trigger the focus styles on the element we should call `element.focus()`.

Reply:
Should we keep [the original `focus` styles](https://github.com/WordPress/gutenberg/pull/32250/commits/e5f3cb681c3939416bfccd41e058361c93fadc5c) though, and just add them under a `&:focus { ... }` css selector?

Reply:
Actually, we may not need to  — there are already some focus styles included in the rest of the component.